sherpa/plot/backends.py

```diff
@@ -0,0 +1,961 @@
+              color=None,
+              linestyle='solid',
+              linewidth=None,
+              marker='None',
+              alpha=None,
+              markerfacecolor=None,
+              markersize=None,
+              ecolor=None,
+              capsize=None,
+              # barsabove=False,
+              **kwargs):
+        """Draw histogram data.
+
+        The histogram is drawn as horizontal lines connecting the
+        start and end points of each bin, with vertical lines connecting
+        consecutive bins. Non-consecutive bins are drawn with a
+        (NaN, NaN) between them so no line is drawn connecting them.
+
+        Points are drawn at the middle of the bin, along with any
+        error values.
+
+        .. warning::
+           This backend is a non-functional dummy. The documentation is provided
+           as a template only.
+
+        Parameters
+        ----------
+        x0 : array-like or scalar number
+            lower bin boundary values
+        x1 : array-like or scalar number
+            upper bin boundary values
+        y : array-like or scalar number
+            y values, same dimension as `x0`.
+        {kwargs}
+        """
+        pass
+
+    @add_kwargs_to_doc(kwargs_doc)
+    @translate_args
+    def contour(self, x0, x1, y, *,
+                levels=None,
+                title=None, xlabel=None, ylabel=None,
+                overcontour=False, clearwindow=True,
+                xlog=False, ylog=False,
+                label=None,
+                colors=None,
+                linestyles='solid',
+                linewidths=None,
+                alpha=None,
+                **kwargs):
+        """Draw 2D contour data.
+
+        .. warning::
+           This backend is a non-functional dummy. The documentation is provided
+           as a template only.
+
+        Parameters
+        ----------
+        x0 : array-like
+            independent axis in the first dimenation (on regular grid, flattened)
+        x1 : array-like
+            independent axis in the second dimenation (on regular grid, flattened)
+        y : array-like
+            dependent axis (i.e. image values) (on regular grid, flattened)
+        {kwargs}
+        """
+        pass
+
+    @translate_args
+    def vline(self, x, *,
+              ymin=0, ymax=1,
+              title=None, xlabel=None, ylabel=None,
+              overplot=False, clearwindow=True,
+              linecolor=None,
+              linestyle=None,
+              linewidth=None,
+              **kwargs):
+        """Draw a vertical line
+
+        .. warning::
+           This backend is a non-functional dummy. The documentation is provided
+           as a template only.
+
+        Parameters
+        ----------
+        x : float
+            x position of the vertical line in data units
+        ymin, ymax : float
+            beginning and end of the vertical line in axes coordinates, i.e. from
+            0 (bottom) to 1 (top).
```

Comment:
You are correct. The plotting backend underneath (matplotlib chips, bokeh) does of course have separation between data coordinates, aixs coordinates, figure coordinates, and possibly other coordinate systems. However, sherpa itself did not need that in the past, except in the have of hline/vline, which means "draw a line at this y (hline)/x (vline) value form one end of the plot to the other. In the plots we have right now, that's used to create an annotation, e.g. ratio plots mark the y=1 line. 

So far, that's all we needed and I have not changed that - most of the code in this module looks new because I moved module functions into class methods, so the order and indentation is new, but not the code itself.

Now, I *could* image plot types that want more sophisticated "annotations" or "markup" but that's beyond the scope if this PR and in #1375 we said that we would add further annotations as needed; with the system of a base class here that implements any new function as a no-op we can do that without breaking backwards compatibility.